src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenIfOperator.vb

```diff
@@ -1247,5 +1247,105 @@ Friend Module BIFOpResult0011mod
 }]]>)
         End Sub
 
+        <Fact>
+        Public Sub IfOnUnconstrainedMethodTypeParameter()
+            CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Friend Module Mod1
+        Sub M1(Of T)(arg1 As T, arg2 As T)
+            System.Console.WriteLine(If(arg1, arg2))
```

Comment:
>If(arg1, arg2) [](start = 37, length = 14)

It would be good to have a test for a scenario when there is a conversion for the test expression

Reply:
Thanks, discovered an issue in Emit with this test.

---
In reply to: [217217442](https://github.com/dotnet/roslyn/pull/29724#discussion_r217217442) [](ancestors = 217217442)

Reply:
> discovered an issue in Emit with this test.

Do we have similar test for C#?

---
In reply to: [217544873](https://github.com/dotnet/roslyn/pull/29724#discussion_r217544873) [](ancestors = 217544873,217217442)

Reply:
Yes, [https://github.com/dotnet/roslyn/blob/features/null-operator-enhancements/src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOperators.cs#L1388](https://github.com/dotnet/roslyn/blob/features/null-operator-enhancements/src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOperators.cs#L1388)

---
In reply to: [217546650](https://github.com/dotnet/roslyn/pull/29724#discussion_r217546650) [](ancestors = 217546650,217544873,217217442)